x/staking/handler.go

```diff
@@ -180,6 +182,16 @@ func handleMsgEditValidator(ctx sdk.Context, msg types.MsgEditValidator, k keepe
 		validator.Commission = commission
 	}
 
+	if msg.MinSelfBond != nil {
+		if !(*msg.MinSelfBond).GT(validator.MinSelfBond) {
```

Comment:
Why are we taking pointers here? doesn't make sense

Reply:
++

I Know we have to do this for commission because of the way JSON serialization works for ints, but since min self-bond value can never be zero, I don't think we need a pointer here.

Reply:
It's to deal with if the user doesn't want to change the value.

There's no nil value for sdk.Int, and so if they keep it empty, it will set msg.MinSelfBond to 0 by default, and then fail ValidateBasic().

Reply:
Oh, Bez I think I see what you mean.  Because it should never be settable to 0, override the 0 value and have it mean "keep as is".

Problem is from when I was trying to do the `MaxDecreaseRate` thing.  That actually has a meaningful 0 value that is settable, so that trick won't work.  And because we may want to reintroduce `MaxDecreaseRate` at some point, I think its best to keep the current pointer method as is.